app/src/main/java/com/marcelorcorrea/falae/activity/DisplayActivity.kt

```diff
@@ -58,6 +64,16 @@ class DisplayActivity : AppCompatActivity(), PageFragment.PageFragmentListener,
         overridePendingTransition(R.anim.enter_from_left, R.anim.exit_to_right)
     }
 
+    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
+        if (scanMode) {
+            if (keyCode == KeyEvent.KEYCODE_HEADSETHOOK) {
+                val pageFragment = supportFragmentManager.fragments.lastOrNull() as PageFragment?
```

Comment:
Maybe add a predicate to lastOrNull to make sure that the last fragment is actually class of PageFragment.
"fragments.lastOrNull { it is PageFragment}"